dnf5/commands/download/download.cpp

```diff
@@ -102,18 +102,21 @@ void DownloadCommand::run() {
     auto & ctx = get_context();
 
     std::vector<libdnf::rpm::Package> download_pkgs;
-    libdnf::rpm::PackageSet result_pset(ctx.base);
+    libdnf::rpm::PackageQuery result_query =
+        libdnf::rpm::PackageQuery(ctx.base, libdnf::sack::ExcludeFlags::APPLY_EXCLUDES, true);
     libdnf::rpm::PackageQuery full_package_query(ctx.base);
     for (auto & pattern : *patterns_to_download_options) {
         libdnf::rpm::PackageQuery package_query(full_package_query);
         auto option = dynamic_cast<libdnf::OptionString *>(pattern.get());
         package_query.resolve_pkg_spec(option->get_value(), {}, true);
-        result_pset |= package_query;
+        result_query |= package_query;
     }
+    result_query.filter_priority();
+    result_query.filter_latest_evr();
```

Comment:
I think this filters need to be part of each of pattern queries and doing this for the accumulated query is not enough in case you want download multiple versions of the package:

```
$ dnf5 repoquery dnf5
dnf5-0:5.0.7-20230411005321.113.g99f6bc40.fc37.x86_64
dnf5-0:5.0.8-0.20230414005203.5.0.7+117.gd2a9d121.fc37.x86_64
dnf5-0:5.0.9-20230419005437.0.gf7eea13f.fc37.x86_64

$ dnf5 download dnf5-0:5.0.7-20230411005321.113.g99f6bc40.fc37.x86_64 dnf5-0:5.0.8-0.20230414005203.5.0.7+117.gd2a9d121.fc37.x86_64
Downloading Packages:
 dnf5-0:5.0.9-20230424005453.22.g41676c39.fc37.x86_64
 dnf5-0:5.0.9-20230424005453.22.g41676c39.fc37.src
```

So only the latest version of the required packages got downloaded (and even version that was not requested). The other issue is that although only `x86_64` arch was required, also `src` package is downloaded. But that's probably out of scope of this PR.

Reply:
Hmm, I didn't think of downloading multiple versions of the same package explicitly... Probably better to track it in the new issue?

Reply:
I created a small update of the PR - https://github.com/rpm-software-management/dnf5/pull/488

Reply:
Thanks, didn't consider that case!